lib/util/http/github.ts

```diff
@@ -273,12 +278,33 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
     options?: InternalHttpOptions & GithubHttpOptions,
     okToRetry = true
   ): Promise<HttpResponse<T>> {
-    const opts = {
+    const opts: GithubHttpOptions = {
       baseUrl,
       ...options,
       throwHttpErrors: true,
     };
 
+    if (!opts.token) {
+      const authUrl = new URL(resolveBaseUrl(opts.baseUrl!, url));
+
+      if (opts.repository) {
+        // set authUrl to https://api.gihub.com/repos/org/repo or https://gihub.domain.com/api/v3/repos/org/repo
+        authUrl.hash = '';
+        authUrl.search = '';
+        authUrl.pathname = joinUrlParts(
+          authUrl.pathname.startsWith('/api/v3') ? '/api/v3' : '',
```

Comment:
What scenario does it not start with `/api/v3` ?

Reply:
i think for github.com?, see urls above in comment